### PR TITLE
Bug/proptype patch

### PR DESCRIPTION
**Pull request by huayunh**
Changes proposed in this Pull Request:
- Fix prop types
- Fix issues with statusColor & icons caused by forced type cast
- Allow statusColor for nested items

### CODE REVIEW

**Comment by huayunh** [comment]
Taking that two slider out to avoid breaking the code... besides, I think those two might not be that critical to demonstrate our api anyway...

File: docs/Drawer.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ The `items` prop of the `DrawerNavGroup` takes a list of items with the followin
 | title           | The text to show on the first line                                                                             | `string`           | yes      |                                  |    
 
 #### NestedNavItem Object
-The `items` property of the NavItem can be nested to create a tree structure with expand/collapse panels. Nested items take the same properties as `NavItem` with the exception of `icon` (nested items cannot use icons).
+The `items` property of the NavItem can be nested to create a tree structure with expand/collapse panels. Nested items take the same properties as `NavItem` with the exception of `icon` (nested items cannot use icons). Besides, the default `expandIcon` prop is `<ArrowDropUp />`.
```
Comment (joebochill) [inline docs/Drawer.md]: We don't need to call out the default icon here...they'll see it when they use the component. Or if you really want to tell them what it is, you can call it out in the API row above.

File: components/src/core/Drawer/DrawerNavGroup.tsx [hunk 2]
```diff
@@ -407,25 +408,37 @@ export const DrawerNavGroup: React.FC<DrawerNavGroupProps> = (props) => {
...
 DrawerNavGroup.propTypes = {
     activeBackgroundColor: PropTypes.string,
     activeFontColor: PropTypes.string,
     activeIconColor: PropTypes.string,
+    activeItem: PropTypes.string,
+    activeBackgroundShape: PropTypes.oneOf(['round', 'square']),
     backgroundColor: PropTypes.string,
     chevron: PropTypes.bool,
-    titleContent: PropTypes.element,
+    titleContent: PropTypes.node,
```
Comment (joebochill) [inline components/src/core/Drawer/DrawerNavGroup.tsx]: Does this work correctly if you pass in a node instead of an element?

Comment (huayunh) [reply components/src/core/Drawer/DrawerNavGroup.tsx]: titleContent has been a ReactNode type when it's declared in type DrawerNavGroupProps. But I tried in the showcase to pass in a `<></>`, and it works, so I think this change is legit.

**Pull request merged**